docs/source/software_process.rst

```diff
@@ -0,0 +1,64 @@
+..  _software_process:
+
+================
+Software Process
+================
+
+----------------------
+Continuous Integration
+----------------------
+
+PyNWB is tested against Ubuntu, macOS and Windows operating systems.
+The project has both unit and integration tests.
+
+* Circleci runs all PyNWB tests against Ubuntu
+* Appveyor runs all PyNWB tests against Windows
+* Travis runs all PyNWB tests against macOS
+
+Tests are ran everytime there is a new branch. That way the contributor knows
+that they are not breaking core functionality of PyNWB on all operating systems.
+
+There are badges in the readme_ file which shows the current condition of the dev branch.
+
+.. _readme: https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/README.rst
+
+--------
+Coverage
+--------
+
+Coverage is computed and reported using the coverage_ tool. There is a badge in the readme file which
+shows percentage coverage. A detailed report can be found on codecov_ which shows line by line which
+lines are covered by the tests.
+
+.. _coverage: https://coverage.readthedocs.io
+.. _codecov: https://codecov.io/gh/NeurodataWithoutBorders/pynwb/tree/dev/src/pynwb
+
+------------
+Requirements
+------------
+
+Requirements of the project can be created and captured by following the code snippet below:
```

Comment:
May be worth describing difference between dependencies in setup.py and requirements.txt adding relevant references.